src/openloco/windows/townwnd.cpp

```diff
@@ -1,31 +1,863 @@
+
+                    const uint32_t ebx = (town->var_38 >> 3) + 5;
+                    const int16_t currentYear = current_year();
+                    int16_t tempYear = currentYear - 51;
+                    set_current_year(tempYear);
+
+                    for (uint8_t i = 8; i > 0; i--)
+                    {
+                        for (uint32_t j = ebx; j > 0; j--)
+                        {
+                            registers regs;
+                            regs.esi = (int32_t)town;
+                            regs.eax = 0xFF;
+
+                            call(0x00498116, regs);
+                            call(0x004975E0, regs);
+                        }
+
+                        tempYear += 7;
+                        set_current_year(tempYear);
+                    }
+
+                    set_current_year(currentYear);
+
+                    // Set new history entry.
+                    uint8_t historyEntry = std::min<uint8_t>(town->population / 50, 255);
+                    town->history[town->history_size - 1] = historyEntry;
+
+                    // Play construction sound at the town centre.
+                    int16_t tileZ = openloco::map::tile_element_height(town->x, town->y) & 0xFFFF;
+                    audio::play_sound(audio::sound_id::construct, loc16(town->x + 16, town->y + 16, tileZ));
+                    break;
+                }
+
+                case widx::demolish_town:
+                {
+                    loco_global<string_id, 0x009C68E8> gameErrorString;
+                    gameErrorString = string_ids::cant_remove_town;
+                    bool success = game_commands::do_50(1, self->number);
+                    if (!success)
+                        break;
+
+                    loco_global<uint16_t, 0x009C68E0> gameCommandMapX;
+                    loco_global<uint16_t, 0x009C68E2> gameCommandMapY;
+                    loco_global<uint16_t, 0x009C68E4> gameCommandMapZ;
+
+                    audio::play_sound(audio::sound_id::demolish, loc16(gameCommandMapX, gameCommandMapY, gameCommandMapZ));
+                    break;
+                }
+            }
+        }
+
+        static void initViewport(window* self);
+
+        // 0x004993A5
+        static void on_resize(window* self)
         {
-            w->enabled_widgets &= ~(1 << 1);
+            // Call to sub_498E9B has been deliberately omitted.
+
+            self->set_size(gfx::ui_size_t(192, 161), gfx::ui_size_t(600, 440));
+
+            common::on_resize(self);
+
+            if (self->viewports[0] != nullptr)
+            {
+                uint16_t newWidth = self->width - 30;
+                if (!is_editor_mode())
+                    newWidth += 22;
+
+                uint16_t newHeight = self->height - 59;
+
+                auto& viewport = self->viewports[0];
+                if (newWidth != viewport->width || newHeight != viewport->height)
+                {
+                    viewport->width = newWidth;
```

Comment:
This will all get reset when `initViewport` gets called after doing `self->saved_view.clear();`